EventHandler.java

```diff
@@ -0,0 +1,166 @@
+ 
+        final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS; 
+        final ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber = new ConcurrentHashMap<String, Pair<Long, String>>(); 
+        final ConcurrentHashMap<String, List<KeyVersion>> filesToDelete = new ConcurrentHashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+        bucketsToProcess.parallelStream().forEach(bucketName -> { 
+            final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber); 
+            filesToDelete.put(bucketName, filesProcessed); 
+    }); 
+         
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        //Create a new file in the Constants.SUMMARY_BUCKET 
+        logger.log("Map of statuses -> " + latestStatusForTrackingNumber); 
+        String summaryUpdateName = Long.toString(System.currentTimeMillis()); 
+         
+        EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+     
+        Waiter waiter = EventHandler.getS3Client().waiters().objectExists();
+        try{
+            waiter.run(new WaiterParameters<>(new GetObjectMetadataRequest(Constants.SUMMARY_BUCKET, summaryUpdateName)));
+            deleteProcessedFiles(filesToDelete);
+            logger.log("All updates successfully processed");
+        } catch (WaiterTimedOutException e){
+            throw new RuntimeException("Failed to write sumary status, will be retried in 15 minutes");
+        }
+    } 
+ 
+    private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber) { 
+ 
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        logger.log("Processing Bucket: " + bucketName); 
+ 
+        ObjectListing files = s3Client.listObjects(bucketName); 
+        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>(); 
+        int iteration = 0;
+        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) { 
+            S3ObjectSummary summary = (S3ObjectSummary) iterator.next(); 
+            logger.log("Reading Object: " + summary.getKey()); 
+ 
+            String trackingNumber = summary.getKey().split("--")[0]; 
+            Pair<Long, String> lastKnownStatus = latestStatusForTrackingNumber.get(trackingNumber); 
+ 
+            // Check if this shipment has already been delivered, skip this file 
+            if (lastKnownStatus != null && "DELIVERED".equals(lastKnownStatus.getRight())) { 
+                continue; 
+            } 
+ 
+            String fileContents = s3Client.getObjectAsString(bucketName, summary.getKey()); 
+ 
+            if (!isValidFile(fileContents)) { 
+                logger.log(String.format("Skipping invalid file %s in %s itr", summary.getKey(), iteration)); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: iteration (declared at line 95). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 111. Consider formatting this data with &#39;%d&#39; instead.